packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -315,7 +316,19 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       if (typeof ref === 'function') {
         ref(instanceToUse);
       } else {
-        ref.value = instanceToUse;
+        if (__DEV__) {
+          if (!ref.hasOwnProperty('current')) {
+            warning(
+              false,
+              'Unexpected ref object provided for %s. ' +
+                'Use either a ref-setter function or Reacte.createRef().%s',
```

Comment:
@bvaughn `Reacte` is a typo right 🤔

Reply:
Haha yeah. I think that was fixed in master but would appreciate a double check

Reply:
😁 Yeah.